File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int ans=1,t=1;
        for(int i=1;i<nums.size();i++){
            if(nums[i-1]+1==nums[i]) t++;
            else if(nums[i]==nums[i-1]) continue;
            else t=1;
            ans=max(ans,t);
        }
        return nums.size()==0?0:ans;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length==0) return 0;
        Set<Integer> number = new HashSet<>();
        for(int i : nums){
            number.add(i);
        }
        int longest=1;
        for(int i : nums){
            if(!number.contains(i-1)){
                int curr_longest=0;
                int curr_number=i;
                while(number.contains(curr_number)){
                    curr_longest++;
                    curr_number++;
                }
                if(curr_longest>longest){
                    longest=curr_longest;
                }
            }
        }
        return longest;
    }
}